File: backends/php-laravel/app/Domain/Client/Mappers/ClientApiMapper.php
<?php
/**
 * This is a program developed by BobTabo.
 *
 * Copyright (c) 2026 BobTabo. All Rights Reserved.
 */
namespace App\Domain\Client\Mappers;

use App\Domain\Client\Entities\Client;
use App\Domain\Client\Enums\ClientStatus;

/**
 * クライアント Entity を API レスポンス向け配列へ変換します。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 * @package App\Domain\Client\Mappers
 */
class ClientApiMapper
{
    /**
     * @return array<string, mixed>
     */
    public static function toResponseArray(Client $client): array
    {
        return [
            'id' => $client->id,
            'name' => $client->name,
            'identifier' => $client->identifer,
            'post_code' => $client->postCode,
            'pref' => $client->pref,
            'city' => $client->city,
            'address' => $client->address,
            'building' => $client->building,
            'tel' => $client->tel,
            'email' => $client->email,
            'status' => $client->status instanceof ClientStatus ? $client->status->value : $client->status,
            'start_at' => $client->startAt?->toIso8601String(),
            'stop_at' => $client->stopAt?->toIso8601String(),
            'created_at' => $client->createdAt?->toIso8601String(),
            'updated_at' => $client->updatedAt?->toIso8601String(),
            'version' => $client->version,
        ];
    }
}

File: backends/java-spring/src/main/java/com/authorization/domain/client/mappers/ClientApiMapper.java
/**
 * This is a program developed by BobTabo.
 *
 * Copyright (c) 2026 BobTabo. All Rights Reserved.
 */
package com.authorization.domain.client.mappers;

import com.authorization.domain.client.entities.Client;
import com.authorization.domain.client.valueobjects.ClientDetailVo;
import com.authorization.domain.client.valueobjects.ClientStoreVo;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;

/**
 * クライアント Entity を API 向け ValueObject へ変換する MapStruct マッパーです。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 */
@Mapper(componentModel = "spring")
public interface ClientApiMapper {

    /**
     * クライアント Entity を詳細 ValueObject に変換します。
     *
     * @param client クライアント Entity
     * @return クライアント詳細 ValueObject
     */
    @Mapping(target = "status", expression = "java(client.getStatus() == null ? null : client.getStatus().value())")
    ClientDetailVo toDetailVo(Client client);

    /**
     * クライアント Entity を登録・更新結果 ValueObject に変換します。
     *
     * @param client クライアント Entity
     * @return クライアント登録・更新 ValueObject
     */
    @Mapping(target = "status", expression = "java(client.getStatus() == null ? null : client.getStatus().value())")
    ClientStoreVo toStoreVo(Client client);
}
